### ISSUE TITLE
Additional plotting style

### ISSUE THREAD

**Issue by MdKquantum** [toolsforexperiments/plottr#180]
Hello,

I was working with some data in plottr lately and missed a way to show a 2D dataset as the different linecuts. I know you can plot the linecuts one by one, however for some data it is more illustrative to show all the linecuts in the same plotwindow.  I wrote a simple function and implementation of these kind of plots, see below:

```
def plot_as_linecuts(ax: Axes, x: np.ndarray, y: np.ndarray,
                     z: np.ndarray, cmap, **kw):
    
    im = "linecuts"
    sm = plt.cm.ScalarMappable(cmap=cmap, norm=plt.Normalize(vmin=np.amin(y), vmax=np.amax(y)))
    
    if len(x.shape) == 1:
        ax.plot(x,z,color = sm.to_rgba((y[0,0])),**kw)
    else:
        for i in range(x.shape[1]):
            ax.plot(x[:,i],z[:,i],color = sm.to_rgba((y[0,i])),**kw)
    
    return im, sm
```

It is very simple and can probably made a bit smarter by adding masked arrays to show continuous progress, now it just plots a new linecut after a step on the slow axis.

I was wondering if others are interested in this extension as an addition to the current plottypes and if so would it be possible to add something similar to this to a future release of plottr. If this is something I should keep in my own fork please let me know then I can close this topic.

**Comment by jenshnielsen** [comment]
I personally think it sounds useful. If you add an additional style that can be selected when its useful that sounds like a great win to everyone

**Comment by MdKquantum** [comment]
Yes that is how I currently have it set-up. I just added an additional style in the PlotStyle enumerator, and then a button in the toolbar which activates the above function. 

I can implement it, but I'm new to github so would probably need some help with how github exactly works.

**Comment by Akshita07** [comment]
Hi @MdKquantum, I am here to help you with any github questions you have.

**Comment by Akshita07** [comment]
To be able to contribute to this repository. You first need to fork it on your account. Then setup a local working environment by cloning the forked repository on to your PC by using command `git clone <forked-repo-url>` . Please let me know when you have done this and we can then move forward.

**Comment by MdKquantum** [comment]
I have it forked and cloned to my pc now. 

As a sidenote I have python installed via anaconda.

**Comment by Akshita07** [comment]
Ok, that is good. Now you can add source repo as upstream remote to your cloned repo by using command `git remote add upstream <source-repo-url>` . 

On a side note, you will need a conda environment to test your changes.
You can create a conda dev environment by running command `conda create -n plottr-dev python=3.7.9`. It will create an conda environment with name plottr-dev. Activate this environment with command `conda activate plottr-dev`.  And then install your plottr from the git repo in editable version by using command `pip install -e .` 

Coming back to git.
Once the upstream is set, you can create a new branch where you will add all your changes by running command `git checkout -b <branch-name>` . You can check which branch you are working on by command `git branch`. Once you are done with all your changes on the branch you can try to see those by running `plottr-inspectr` in the environment we created above. 
You can commit these changes with command `git commit -m "changes"` when you are satisfied.
After that, push the branch on your forked repo by `git push origin <branch-name>`.
Now from the github account you can create a pull request for this branch to be merged into the source code.

Please follow these instructions, and let me know if you face any problems anywhere.

**Comment by MdKquantum** [comment]
I just put in a pull request. Thank you for your help @Akshita07 !

**Comment by Akshita07** [comment]
@MdKquantum Thank you for your PR. One quick thing I noticed which I missed and would like to mention you is while committing the changes in future with command `git commit -m "changes"` you should replace the word 'changes' with meaningful short comments which mention the changes you have made in the code. It will help the developers understand the changed code better.
